tinygrad/tensor.py

```diff
@@ -301,16 +301,29 @@ def pad(self, arg:Tuple[Optional[Tuple[int, int]], ...], value:float=0.0) -> Ten
   #        - if first Tensor passed in (expand dims) is not at dim 0
   #        - and following Tensors does not follow consecutively to the end of fancy indexing's dims
   def __getitem__(self, indices) -> Tensor: # indices: Union[int, slice, Tensor, None, Ellipsis, List, Tuple[Union[int, slice, Tensor, None, Ellipsis], ...]]
+    def validate_slice(sl: slice):
```

Comment:
a general direction i want to split these and make it more manageable is (1) input normalization and validation (2) basic indexing, where we return a tensor with the same base as input (like a torch view), (3) advanced indexing, where we return a copy. think about this separation and add helper functions where you see fit